alchemy/src/docker/container.ts

```diff
@@ -0,0 +1,267 @@
+   */
+  internal: number | string;
+
+  /**
+   * Protocol (tcp or udp)
+   */
+  protocol?: "tcp" | "udp";
+}
+
+/**
+ * Volume mapping configuration
+ */
+export interface VolumeMapping {
+  /**
+   * Host path
+   */
+  hostPath: string;
+
+  /**
+   * Container path
+   */
+  containerPath: string;
+
+  /**
+   * Read-only flag
+   */
+  readOnly?: boolean;
+}
+
+/**
+ * Network mapping configuration
+ */
+export interface NetworkMapping {
+  /**
+   * Network name or ID
+   */
+  name: string;
+
+  /**
+   * Aliases for the container in the network
+   */
+  aliases?: string[];
+}
+
+/**
+ * Properties for creating a Docker container
+ */
+export interface ContainerProps {
+  /**
+   * Image to use for the container
+   * Can be an Alchemy Image or RemoteImage resource or a string image reference
+   */
+  image: Image | RemoteImage | string;
+
+  /**
+   * Container name
+   */
+  name?: string;
+
+  /**
+   * Command to run in the container
+   */
+  command?: string[];
+
+  /**
+   * Environment variables
+   */
+  environment?: Record<string, string>;
+
+  /**
+   * Port mappings
+   */
+  ports?: PortMapping[];
+
+  /**
+   * Volume mappings
+   */
+  volumes?: VolumeMapping[];
+
+  /**
+   * Restart policy
+   */
+  restart?: "no" | "always" | "on-failure" | "unless-stopped";
+
+  /**
+   * Networks to connect to
+   */
+  networks?: NetworkMapping[];
+
+  /**
+   * Whether to remove the container when it exits
+   */
+  removeOnExit?: boolean;
+
+  /**
+   * Start the container after creation
+   */
+  start?: boolean;
+}
+
+/**
+ * Docker Container resource
+ */
+export interface Container
+  extends Resource<"docker::Container">,
+    ContainerProps {
+  /**
+   * Container ID
+   */
+  id: string;
+
+  /**
+   * Container state
+   */
+  state?: "created" | "running" | "paused" | "stopped" | "exited";
+
+  /**
+   * Time when the container was created
+   */
+  createdAt: number;
+}
+
+/**
+ * Create and manage a Docker Container
+ *
+ * @example
+ * // Create a simple Nginx container
+ * const webContainer = await Container("web", {
+ *   image: "nginx:latest",
+ *   ports: [
+ *     { external: 8080, internal: 80 }
+ *   ],
+ *   start: true
+ * });
+ *
+ * @example
+ * // Create a container with environment variables and volume mounts
+ * const appContainer = await Container("app", {
+ *   image: customImage, // Using an Alchemy RemoteImage resource
+ *   environment: {
+ *     NODE_ENV: "production",
+ *     API_KEY: "secret-key"
+ *   },
+ *   volumes: [
+ *     { hostPath: "./data", containerPath: "/app/data" }
+ *   ],
+ *   ports: [
+ *     { external: 3000, internal: 3000 }
+ *   ],
+ *   restart: "always",
+ *   start: true
+ * });
+ */
+export const Container = Resource(
+  "docker::Container",
+  async function (
+    this: Context<Container>,
+    id: string,
+    props: ContainerProps,
+  ): Promise<Container> {
+    // Initialize Docker API client
+    const api = new DockerApi();
```

Comment:
do we need to support overriding any of the auth/connection details (e.g. how to connect to a local vs remote daemon).

I recall doing interesting things with build kit that required specifying some of these details.

Reply:
I have limited bandwidth now. I would leave this for next iteration or for someone else to pick up.